timeboost/db.go

```diff
@@ -129,6 +129,34 @@ func (d *SqliteDatabase) InsertBid(b *ValidatedBid) error {
 	return nil
 }
 
+func (d *SqliteDatabase) GetBids(maxDbRows int) ([]*SqliteDatabaseBid, uint64, error) {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	var maxRound uint64
+	query := `SELECT MAX(Round) FROM Bids`
+	err := d.sqlDB.Get(&maxRound, query)
+	if err != nil {
+		return nil, 0, fmt.Errorf("failed to fetch maxRound from bids: %w", err)
+	}
+	var sqlDBbids []*SqliteDatabaseBid
+	if maxDbRows == 0 {
+		if err := d.sqlDB.Select(&sqlDBbids, "SELECT * FROM Bids WHERE Round < ? ORDER BY Round ASC", maxRound); err != nil {
+			return nil, 0, err
+		}
+		return sqlDBbids, maxRound, nil
+	}
+	if err := d.sqlDB.Select(&sqlDBbids, "SELECT * FROM Bids WHERE Round < ? ORDER BY Round ASC LIMIT ?", maxRound, maxDbRows); err != nil {
```

Comment:
I think it would be worth adding an index on Round now that we're sorting and filtering on it. In schema.go you can add the index:

```
	version1 = `
CREATE TABLE IF NOT EXISTS Bids (
    Id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
    ChainId TEXT NOT NULL,
    Bidder TEXT NOT NULL,
    ExpressLaneController TEXT NOT NULL,
    AuctionContractAddress TEXT NOT NULL,
    Round INTEGER NOT NULL,
    Amount TEXT NOT NULL,
    Signature TEXT NOT NULL
);

CREATE INDEX idx_bids_round ON Bids(Round);
`
```

Reply:
Cool, added this